it/connectors/FinancialDataConnectorISpec.scala

```diff
@@ -0,0 +1,134 @@
+
+import scala.concurrent.ExecutionContext
+
+class FinancialDataConnectorISpec extends ComponentSpecBase {
+
+  val connector: API1811.FinancialDataConnector = new API1811.FinancialDataConnector(httpClient, appConfig)
+  implicit val hc: HeaderCarrier = HeaderCarrier()
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  val regimeType = "VATC"
+  val invalidRegimeType = "9"
+  val vatRegime: TaxRegime = VatRegime(id = "12345678")
+  val queryParameters: FinancialDataQueryParameters = FinancialDataQueryParameters()
+
+  def generateUrl(regimeType: String): String =  s"/penalty/financial-data/VRN/12345678/$regimeType" +
+    s"?onlyOpenItems=false&includeLocks=true&calculateAccruedInterest=true&removePOA=true&customerPaymentInformation=true"
+
+  "getFinancialData" should {
+
+    "return a FinancialTransactionsModel" when {
+
+      s"an $OK response is received from financial transactions and the response can be parsed" in {
+
+        stubGetRequest(
+          generateUrl(regimeType),
+          OK,
+          fullFinancialTransactionsJson.toString()
+        )
+
+        val expectedResult = fullFinancialTransactions
+
+        val result: FinancialTransactionsResponse =
+          await(connector.getFinancialData(vatRegime,queryParameters))
+        result shouldBe Right(expectedResult)
+
+      }
+
+      "return an ErrorModel" when {
+
+        s"a $OK response is received from financialTransactions, but the response cannot be parsed" in {
+
+          stubGetRequest(
+            generateUrl(regimeType),
+            OK,
+            """{"foo":"bar"}"""
+          )
+
+          val expectedResult = Left(Error(BAD_REQUEST,
+            "UNEXPECTED_JSON_FORMAT - The downstream service responded with json which did not match the expected format."))
+
+          val result: FinancialTransactionsResponse =
+            await(connector.getFinancialData(vatRegime, queryParameters))
+          result shouldBe expectedResult
+
+        }
+        s"a $BAD_REQUEST response is received from financial transactions" in {
+
+          stubGetRequest(
+            generateUrl(invalidRegimeType),
+            BAD_REQUEST,
+            Json.obj("code" -> 400,
+              "reason" -> "BAD REQUEST").toString()
+          )
+
+            val expectedResult = Left(Error(BAD_REQUEST,
+              "UNEXPECTED_JSON_FORMAT - The downstream service responded with json which did not match the expected format."))
+
+          val result: FinancialTransactionsResponse =
+            await(connector.getFinancialData(vatRegime, queryParameters))
+          result shouldBe expectedResult
+        }
+        s"a $NOT_FOUND response is received from Financial Transactions" in {
+
+          stubGetRequest(
+            generateUrl(regimeType),
+            NOT_FOUND,
+            Json.obj("code" -> 404,
+              "reason" -> "A bad request has been made, this could be due to one or more issues with the request").toString()
```

Comment:
Doesn't exactly matter as it's just a made up error reason but this sentence itself is a bit confusing as it suggests this is a bad request scenario when it's actually just 404 no data found